screenlog.js

```diff
@@ -1,69 +1,104 @@
 (function () {
-
+	
 	var logEl,
 		isInitialized = false,
 		_console = {}; // backup console obj to contain references of overridden fns.
-
-	function createElement( tag, css ) {
-		var element = document.createElement( tag );
+		_options = {
+						bgColor: 'black',
+						logColor: 'lightgreen',
+						warnColor: 'orange',
+						errorColor: 'red',
+						freeConsole: false,
+						css: '',
+						autoScroll: true
+					};
+	
+	function createElement(tag, css) {
+		var element = document.createElement(tag);
 		element.style.cssText = css;
 		return element;
 	}
-
-	function createPanel(options) {
-		options.bgColor = options.bgColor || 'black';
-		options.color = options.color || 'lightgreen';
-		options.css = options.css || '';
-		var div = createElement( 'div', 'font-family:Helvetica,Arial,sans-serif;font-size:10px;font-weight:bold;padding:5px;text-align:left;opacity:0.8;position:fixed;right:0;top:0;min-width:200px;max-height:50vh;overflow:auto;background:' + options.bgColor + ';color:' + options.color + ';' + options.css);
+	
+	function createPanel() {
+		var div = createElement('div', 'font-family:Helvetica,Arial,sans-serif;font-size:10px;font-weight:bold;padding:5px;text-align:left;opacity:0.8;position:fixed;right:0;top:0;min-width:200px;max-height:50vh;overflow:auto;background:' + _options.bgColor + ';' + _options.css);
 		return div;
 	}
-
-	function log() {
-		var el = createElement( 'div', 'line-height:18px;background:' +
-			(logEl.children.length % 2 ? 'rgba(255,255,255,0.1)' : '')); // zebra lines
-		var val = [].slice.call(arguments).reduce(function(prev, arg) {
-			return prev + ' ' + arg;
-		}, '');
-		el.textContent = val;
-
-		logEl.appendChild(el);
-		// Scroll to last element
-		logEl.scrollTop = logEl.scrollHeight - logEl.clientHeight;
+	
+	function genericLogger(color) {
+		return function() {
+			var el = createElement('div', 'line-height:18px;min-height:18px;background:' +
+				(logEl.children.length % 2 ? 'rgba(255,255,255,0.1)' : '') + ';color:' + color); // zebra lines
+			var val = [].slice.call(arguments).reduce(function(prev, arg) {
+				return prev + ' ' + (typeof arg === "object" ? JSON.stringify(arg) : arg);
+			}, '');
+			el.textContent = val;
+			
+			logEl.appendChild(el);
+			
+			// Scroll to last element, if autoScroll option is set.
+			if(_options.autoScroll) logEl.scrollTop = logEl.scrollHeight - logEl.clientHeight;
+		};
 	}
-
+	
 	function clear() {
 		logEl.innerHTML = '';
 	}
-
-	function init(options){
+	
+	function log() {
+		return genericLogger(_options.logColor).apply(null, arguments);
+	}
+	
+	function warn() {
+		return genericLogger(_options.warnColor).apply(null, arguments);
+	}
+	
+	function error() {
```

Comment:
@vivekannan While you are at this, please also add support for `console.info` (blue color) to completely fix issue #6